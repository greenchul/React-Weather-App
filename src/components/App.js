/* eslint-disable  no-unused-vars */
import "../styles/App.css";
import React, { useState, useEffect } from "react";
import axios from "axios";
import getForecast from "../requests/getForecast";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";

const App = () => {
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });
  const [selectedDate, setSelectedDate] = useState(0);
  const { city, country } = location;

  useEffect(() => {
    getForecast(setForecasts, setLocation, setSelectedDate);
  }, []);

  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };
  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );

  return (
    <div className="App">
      <LocationDetails city={city} country={country} />
      <ForecastSummaries
        forecasts={forecasts}
        onForecastSelect={handleForecastSelect}
      />
      {selectedForecast ? (
        <ForecastDetails forecastDetail={selectedForecast} />
      ) : (
        <div>Select a date</div>
      )}
    </div>
  );
};

export default App;
