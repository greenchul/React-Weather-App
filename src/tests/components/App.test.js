import React from "react";
import { render } from "@testing-library/react";
import App from "../../components/App";

test("it renders correctly", () => {
  const { asFragment } = render(<App />);

  expect(asFragment()).toMatchSnapshot();
});
