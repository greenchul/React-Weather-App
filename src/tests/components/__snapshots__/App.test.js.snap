// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1
      class="location-details"
    >
      , 
    </h1>
    <div
      class="forecast-summaries"
    />
    <div>
      Select a date
    </div>
  </div>
</DocumentFragment>
`;
